helix-core/src/test/java/org/apache/helix/messaging/handling/TestResourceThreadpoolSize.java

```diff
@@ -113,6 +113,35 @@ public void TestThreadPoolSizeConfig() {
     }
   }
 
+  @Test
+  public void testBatchMessageThreadPoolSize() throws InterruptedException {
```

Comment:
Thanks, do you see that this test case fails without my change. Might have to run it multiple times or add a delay in batch message handler before it executes the sub-messages to reproduce it.